rpc/src/v1/types/sync.rs

```diff
@@ -32,14 +33,60 @@ pub struct SyncInfo {
 }
 
 /// Peers info
-#[derive(Default, Debug, Serialize, PartialEq)]
+#[derive(Default, Debug, Serialize)]
 pub struct Peers {
 	/// Number of active peers
 	pub active: usize,
 	/// Number of connected peers
 	pub connected: usize,
 	/// Max number of peers
 	pub max: u32,
+	/// Detailed information on peers
+	pub peers: Vec<PeerInfo>
+}
+
+/// Peer connection information
+#[derive(Default, Debug, Serialize)]
+pub struct PeerInfo {
+	/// Public node id
+	pub id: Option<String>,
+	/// Node client ID
+    pub name: String,
+	/// Capabilities
+	pub caps: Vec<String>, 
+	/// Network information
+	pub network: PeerNetworkInfo,
+	/// Protocols information
+	pub protocols: PeerProtocolsInfo
+}
+
+/// Peer network information
+#[derive(Default, Debug, Serialize)]
+pub struct PeerNetworkInfo {
+	/// Remote endpoint address
+	#[serde(rename="remoteAddress")]
+    pub remote_address: String,
+	/// Local endpoint address
+	#[serde(rename="localAddress")]
+    pub local_address: String
+}
+
+/// Peer protocols information
+#[derive(Default, Debug, Serialize)]
+pub struct PeerProtocolsInfo {
+	/// Ethereum protocol information
+	pub eth: PeerEthereumProtocolInfo
+}
```

Comment:
1) I personally think that { name, props } isn't a good pattern for public APIs - it is hard to document, hard to use (in long term) & hard to maintain. Do you think this will be better than strictly typed struct's for each protocol?
2) as for converting eth to an Option<> - I could do that right now. But when you'll met some code which has return type Option<> and unconditionally returns Some(...) - isn't it a bad pattern?
@rphmeier Could you, please, comment so we could agree on this?